packages/terra-folder-tree/src/subcomponents/FolderTreeItem.jsx

```diff
@@ -98,27 +98,35 @@ const FolderTreeItem = ({
     cx(
       'folder-tree-item',
       { selected: isSelected },
-      { folder: isFolder },
       theme.className,
     ),
   );
 
+  const handleRadioButtonClick = (event) => {
+    // Stop click propagation to prevent triggering expand/collapse when selecting folders
+    event.stopPropagation();
+
+    onClick();
+  };
+
   return (
     <>
       <li
         className={itemClassNames}
         role="treeitem"
         aria-expanded={isFolder ? isExpanded : null}
         aria-selected={isSelected}
-        onClick={onToggle}
-        onKeyDown={onToggle}
+        onClick={isFolder ? onToggle : onClick}
+        onKeyDown={isFolder ? onToggle : onClick}
```

Comment:
Isn't there a specific key or keys that should trigger this action?  I would not expect this to happen for any key pressed.

Reply:
Yes, I took a look at the keyboard Jira (UXPLATFORM-9776) and expand/collapse should be triggered by the right and left arrows and selection should be triggered via the enter key. I went ahead and removed the onKeyDown handler and disabled the eslint error for now, these should be re-enabled as part of the keyboard interaction story.
[ccdc63e](https://github.com/cerner/terra-framework/pull/1937/commits/ccdc63e3b8594b45b7e4d111070b6707dac7135d)